client/network/src/service.rs

```diff
@@ -844,6 +842,32 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		}
 	}
 
+	/// Variation of `request` which takes a sender for sending responses back.
+	///
+	/// Instead of blocking and waiting for a reply, this function returns immediately, sending
+	/// responses via the passed in sender. This alternative API exists to make it easier to
+	/// integrate with message passing APIs.
+	///
+	/// Keep in mind that the connected receiver might receive a `Canceled` event in case of a
+	/// closing connection, this is expected behaviour. In `request` you would get a
+	/// `RequestFailure::Network(OutboundFailure::ConnectionClosed` in that case.
+	pub fn detached_request(
```

Comment:
```suggestion
	pub fn start_request(
```

Reply:
I like that one as well. What do you say @tomaka ?

Reply:
I prefer `detached` but I don't care so much

Reply:
Ok then, I will go with `start_request` as it reminds of some other API with similar semantics.